vpr/src/route/rr_graph_storage.cpp

```diff
@@ -393,6 +393,24 @@ void t_rr_graph_storage::init_fan_in() {
     }
 }
 
+void t_rr_graph_storage::init_fan_in_list() {
+    //Reset all fan-ins to zero
+    edges_read_ = true;
+    node_fan_in_list_.resize(node_storage_.size(), std::vector<RREdgeId>(0));
+    node_fan_in_list_.shrink_to_fit();
+
+    //Walk the graph and increment fanin on all dwnstream nodes
+    for (const auto& edge_id : edge_dest_node_.keys()) {
```

Comment:
If node_fan_in (count of incoming edges) has already been loaded (should have been since we don't appear to have a separate free_ call for it) then you can make this a bit more memory-efficient by getting the right size of vector storage by doing something like the below before the main edge loop that loads the vectors.
```
 for (const auto& edge_id : edge_dest_node_.keys()) {
       node_fan_in_list_[edge_id].reserve(node_fan_in[edge_id];
   }
```